public/app/features/scenes/variables/variants/query/QueryVariable.tsx

```diff
@@ -92,6 +92,14 @@ export class QueryVariable extends MultiValueVariable<QueryVariableState> {
       return of([]);
     }
 
+    // If called for the first time with options already set, return them
+    // TODO: Should we have a new variable refresh option to support skiping initial fetch if options are provided?
+    if (!this._optionsInitialized && this.state.options.length > 0 && this.state.cacheKey) {
+      this._optionsInitialized = true;
+      return of(this.state.options);
+    }
+
+    this._optionsInitialized = true;
```

Comment:
added to query variable, but this needs to be implemented in higher level class for re-use

Reply:
This looks to break any future update (due to dependency change) as this will always return the same options. 

This cannot be done inside getValueOptions I think but in validateAndUpdate in MultiValueVariable, and we need to validate that the current value is found in options.

Reply:
but even there it's tricky, how to not stop it from updating when a dependency changed?

Reply:
Indeed, as I mentioned I think this needs to be moved out to higher level Variable class.